### PR TITLE
Improve linking of event items

### PR DESCRIPTION
**Pull request by koraktor**
Linking is now done using a regular expression trying to find only valid positions of the text we're looking for. This is now also done in a separate method so there's less clutter with duplicate calls.

I know this might not merge cleanly anymore, because you're working on version 2.

### CODE REVIEW

**Comment by koraktor** [comment]
Forgot to mention that this fixes problems like the following:

ios simulator bildschirmfoto 21 07 2013 22 24 35

**Comment by dennisreimann** [comment]
Thanks, I'll need some time to check it, but I'll get back to you :)

**Comment by dennisreimann** [comment]
Very nice, just integrated it into v2. Thanks a lot! :cake:

**Comment by dennisreimann** [comment]
Test

File: Classes/IOCEventCell.m [hunk 1]
```diff
@@ -63,43 +63,58 @@ - (void)setEvent:(GHEvent *)event {
     [self.event addObserver:self forKeyPath:UserGravatarKeyPath options:NSKeyValueObservingOptionNew context:nil];
 	[self setGravatar:gravatar];
     // actions
+    [self linkEventItems];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context {
```
Comment (dennisreimann) [inline Classes/IOCEventCell.m:69]: Test on observeValueForKeyPath

File: Classes/IOCEventCell.m [hunk 2]
```diff
@@ -63,43 +63,58 @@ - (void)setEvent:(GHEvent *)event {
...
+	if ([keyPath isEqualToString:UserGravatarKeyPath] && self.event.user.gravatar) {
+		[self setGravatar:self.event.user.gravatar];
+	}
+}
+
+- (void)setContentText:(NSString *)text {
```
Comment (dennisreimann) [inline Classes/IOCEventCell.m:75]: Test on setContentText

**Comment by dennisreimann** [comment]
Normal comment

**Pull request merged**